exodus/analysis_query/templates/query_upload.html

```diff
@@ -0,0 +1,149 @@
+          <div class="alert alert-light" role="alert">
+            {% trans "The analysis could take a long time, you will be automatically redirected to the report." %}
+          </div>
+        </div>
+      </div>
+    </form>
+  </div>
+</div>
+
+<div class="row justify-content-sm-center mt-5">
+  <div class="col-md-8 col-12 col-centered">
+    <div class="text-center alert alert-light">
+      <p>
+        <a href="{% url 'organization' %}">Exodus Privacy</a>
+        {% trans "is a non-profit organization which provides this service for free." %}
+      </p>
+      <a class="btn btn-outline-primary" href="https://exodus-privacy.eu.org/{{ LANGUAGE_CODE }}/page/contribute/" target="_blank" rel="noreferrer">{% trans "Help us!" %}</a>
+    </div>
+  </div>
+</div>
+{% endblock %}
+
+{% block scripts %}
+<script>
+// Spinner
+var show_spinner=function(){
+  s = document.getElementById("loading")
+  if(s != undefined){
+    s.style.display = "block"
+  }
+}
+jQuery("#handle").on("input", function() {
+    // Extract handle from Google Play url
+    const regex = /id=((?:\w+\.)+\w+)/gmi;
+    const handle = jQuery.trim(jQuery(this).val())
+    jQuery("#reports").hide()
+    if(handle.startsWith("https://play.google.com")){
+        match = regex.exec(handle)
+        while (match != null) {
+          if(match[1]){
+            jQuery(this).val(match[1])
+          }
+          return
+        }
+    } else {
+        const regex_pk = /^((?:\w+\.)+\w+)$/gmi;
+        if(regex_pk.test(handle)){
+            // TODO: Use an API endpoint which returns the latest report
+            jQuery.get("/api/search/"+handle, function(rq){
+                if(rq[handle]){
+                    jQuery("#reports").html('')
+                    jQuery("#reports").show()
+
+                    var p = document.createElement('p')
+                    var name = handle
+                    if(rq[handle].name != '') {
+                        name = rq[handle].name
+                    }
+                    p.innerHTML = "<b>{% trans  'This application has already been analyzed!' %}</b>"
+                    jQuery("#reports").append(p)
+
+                    latest_report = rq[handle].reports.shift()
+                    var report_date = latest_report.creation_date.substring(0, 10)
+                    var p2 = document.createElement('p')
+                    p2.innerHTML = name + ' - {% trans "analyzed on" %} '+report_date+' - <a href="/reports/'+latest_report.id+'/">{% trans "Go to the latest report" %}</a>'
+                    jQuery("#reports").append(p2)
+
+                    var p3 = document.createElement('p')
+                    p3.innerHTML = "{% trans 'If you think εxodus has not yet analyzed the latest version of this application, feel free to submit the analysis.' %}"
+                    jQuery("#reports").append(p3)
+                }
+            })
+        }
+    }
+});
+$(function() {
```

Comment:
Not sure to understand what this whole part is about :D

Reply:
That's the form meant to upload an application. I will cleanup the JS code.